### PR TITLE
Fix(core)  icon change the font size to align ion segment button with icon

### LINKED ISSUE #17069: [ionic4.rc1] [ios] ion-segment-button doesn't align with icon
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
If you have a row of segments with text and one with an icon, the new rc1 (upgraded from beta 13) has very different paddings setup.

One effect of this is that the icon only button is taller than the text only button:

image

**Expected behavior:**

It should align nicely more like this (screenshot from beta 13):

image

**Steps to reproduce:**

Drop the snippet below into an app.

**Related code:**

```
    <ion-segment mode="ios">
          <ion-segment-button value="330" mode="ios">
            <ion-label>330ml</ion-label>
          </ion-segment-button>
          <ion-segment-button value="440" mode="ios">
            <ion-label>440ml</ion-label>
          </ion-segment-button>
          <ion-segment-button value="500" mode="ios">
            <ion-label>500ml</ion-label>
          </ion-segment-button>
          <ion-segment-button value="custom" mode="ios">
            <ion-icon name="create"></ion-icon>
          </ion-segment-button>
    </ion-segment>  
```
Note that wrapping the icon in an `ion-label` does make the row align nicely but makes the icon crazy small:

image

However, the docs have been updated for the `ion-label` wrapper and dont include it around the `ion-icon`:

https://github.com/ionic-team/ionic/blob/master/core/src/components/segment/readme.md

**Other information:**

**Ionic info:** 

```

Ionic:

   ionic (Ionic CLI)             : 4.7.1 (C:\Users\rtpha\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.0.0-rc.1
   @angular-devkit/build-angular : 0.11.4
   @angular-devkit/schematics    : 7.1.4
   @angular/cli                  : 7.1.4
   @ionic/angular-toolkit        : 1.2.2

System:

   NodeJS : v10.13.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10
```

### PR DESCRIPTION
**Pull request by santoshyadav198613**
#### Short description of what this resolves:
fixes #17069 alignment issue  

#### Changes proposed in this pull request:
Changed the font-size and alignment

Before:

Screen Shot 2019-04-25 at 12 29 58 PM
Screen Shot 2019-04-25 at 12 30 05 PM

After:

Screen Shot 2019-04-25 at 12 25 42 PM
Screen Shot 2019-04-25 at 12 25 35 PM

### CODE REVIEW

**Comment by santoshyadav198613** [comment]
Hi @brandyscarney 
Updates with test case please review once.

**Comment by brandyscarney** [comment]
Thanks for the PR and the test! Could you please remove the `mode` attributes though so we can see how it looks in each mode? 🙂 I'll review this soon!

**Comment by santoshyadav198613** [comment]
Thanks @brandyscarney ,
Made the changes, and thanks for code suggestions.

**Comment by santoshyadav198613** [comment]
> In `segment.scss`, could you add the following CSS to the `:host` element:
> 
> ```
> align-self: stretch;
> ```
> and change the height on the `.button-native` class to:
> 
> ```
> height: 100%;
> ```
> This will make it so if the icon size increases the other buttons will increase to fill the segment.

done.

File: core/src/components/segment/segment.scss [hunk 1]
```diff
@@ -11,8 +11,9 @@
   @include font-smoothing();
 
   display: flex;
-
+   
   align-items: center;
```
Comment (brandyscarney) [inline core/src/components/segment/segment.scss]:
```suggestion
   align-items: stretch;
```

File: core/src/components/segment/segment.scss [hunk 2]
```diff
@@ -11,8 +11,9 @@
...
+  align-self : stretch;
```
Comment (brandyscarney) [inline core/src/components/segment/segment.scss]:
```suggestion

```

File: core/src/components/segment/segment.scss [hunk 3]
```diff
@@ -11,8 +11,9 @@
   @include font-smoothing();
 
   display: flex;
-
+
```
Comment (brandyscarney) [inline core/src/components/segment/segment.scss]:
```suggestion

```

Comment (santoshyadav198613) [reply core/src/components/segment/segment.scss]: Done.

File: core/src/components/segment/segment.scss (hunk 2 above)
Comment (santoshyadav198613) [reply core/src/components/segment/segment.scss]: Done.

File: core/src/components/segment/segment.scss (hunk 1 above)
Comment (santoshyadav198613) [reply core/src/components/segment/segment.scss]: Done.

**Comment by santoshyadav198613** [comment]
Done with the changes @brandyscarney .

**Comment by santoshyadav198613** [comment]
Hi @brandyscarney ,
Let me know if any changes are needed.

**Comment by brandyscarney** [comment]
Thanks for the changes! I don't have time to look into it right now, but it looks like indicator on segment is not showing up anymore for MD mode:

| `master` | `branch` |
| ---| ---|
| localhost_3333_src_components_segment_test_basic(Galaxy S5) (1) | localhost_3333_src_components_segment_test_basic(Galaxy S5) |

Could you run it locally to compare? We have some steps on previewing it here: https://github.com/ionic-team/ionic/blob/master/.github/CONTRIBUTING.md#core

**Comment by brandyscarney** [comment]
I had some time to look into this and the fix was actually a bit tricky - I had to update the parent segment button to be `display: flex` in order to make the child button fill the remaining space without bumping the border out of the display. This presented an issue with layout which was actually caused by the `flex-direction` being applied to the wrong component. It all looks good now!

Screen Shot 2019-04-25 at 12 25 35 PM
Screen Shot 2019-04-25 at 12 25 42 PM

**Comment by Huthaifah-kholi** [comment]
so, how we can align the icons to the left/right of label?

**Pull request merged**